### PR TITLE
Add `--action=branch` and various refactors, docs.

### PR DESCRIPTION
**Pull request by ryan-roemer**
Part of #2 mutating features work.
- Implements `--action=branch` to create branches in repositories.
- Add README.md documentation for scenarios.
- Refactor args out to separate file.
- Unify the output display format for all actions.
- Note that there are some `TODO` placeholders for the rest of #2 work that I left. I wanted to tie off just `branch` functionality as I saw this diff was getting pretty big already with the refactor.

Example branching! (On repos with "allow existing" enabled (also by default) since I've been doing lots of test branching).

```
$ bin/multibot.js \
  --gh-token=<SNIPPED> \
  --org FormidableLabs \
  --repos test-repo-o-doom test-repo-o-doom-two \
  --branch=master \
  --branch-new=branch-o-doom \
  --allow-existing=true \
  --action=branch \
  --format=text

###################################
# FormidableLabs/test-repo-o-doom #
###################################
# - Source Branch: master
# - Dest Branch:   branch-o-doom
# - Dest Exists?:  true
# - Commit SHA:    34f93917d20305e0236e83d8af4672c3b97d2b54


#######################################
# FormidableLabs/test-repo-o-doom-two #
#######################################
# - Source Branch: master
# - Dest Branch:   branch-o-doom
# - Dest Exists?:  true
# - Commit SHA:    efdff6c4d7b7acd8e5581123f06416a1ceb3ccd1
```

/cc @boygirl @coopy

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,13 +77,135 @@ A transform is hooked into a multibot action with the option:
...
+
+```sh
+$ multibot \
+  --org FormidableLabs --repos repo1 repo2 repo3 \
+  --transform=foo.js --files README.md \
+  --action=read \
+  --format=diff
+```
+
+<!--
```
Comment (coopy) [inline README.md:94]: TIL GH-flavored Markdown supports HTML comments!

File: lib/repos.js [hunk 2]
```diff
@@ -102,32 +196,167 @@ Repos.prototype.read = function (callback) {
   }, []);
 
   async.map(lookups, function (obj, cb) {
```
Comment (coopy) [inline lib/repos.js:198]: 🌎 GLOBAL: For consistency with other function signatures in this library (e.g. `_checkAndHandleError`), consider renaming `cb` -> `callback`.

Edit: Eh, this is inside of `_checkAndHandleError`. How about `next` here?

File: lib/repos.js [hunk 3]
```diff
@@ -102,32 +196,167 @@ Repos.prototype.read = function (callback) {
...
+  // See: https://developer.github.com/v3/git/refs/#get-a-reference
+  var branchSrcFull = "heads/" + branchSrc;
+  // See: https://developer.github.com/v3/git/refs/#create-a-reference
+  var branchDestFull = "heads/" + branchDest;
+
+  // Although not the most efficient thing to do, we perform all read operations
+  // first to detect and stop on any errors in any branches.
+  async.auto({
+    // Get the most recent commit of the source branch.
+    getSrcRefs: function (cb) {
```
Comment (coopy) [inline lib/repos.js:251]: `cb -> next`?

**Comment by coopy** [comment]
👍  Looks fantastic!

File: lib/repos.js (hunk 2 above)
Comment (ryan-roemer) [inline lib/repos.js:198]: `cb` is the convention for the `async` lib when you have a higher level `callback` for the function as a whole which I have.

`next` is almost never used with `async`, even though it's what we're used to for normal server middleware.

File: lib/repos.js (hunk 3 above)
Comment (ryan-roemer) [inline lib/repos.js:251]: see above.

**Comment by ryan-roemer** [comment]
Thanks @coopy !

**Comment by ryan-roemer** [comment]
Released in `multibot@0.1.0`

**Pull request merged**